tasks/main.yml

```diff
@@ -14,21 +43,25 @@
   command: postconf
   register: __postfix_register_config
   changed_when: false
-  when: postfix_conf | d({}) | length > 0
+  when: postfix_conf | dict2items | rejectattr( 'key', 'match', "previous" )
```

Comment:
should be consistent about quoting here - either use `'` or `"`

also - `match` will match from the beginning of the string - so `"match", "previous"` will match `"previous"`, `"previously"`, `"previousother"`, etc. - if you want to exactly match a string with `match`, better to use `"^previous$"` to make it explicit

Reply:
Thank you, done both.